org.civicrm.reqangular/src/common/services/file-upload.js

```diff
@@ -25,96 +44,76 @@ define([
-              crm_attachment_token: customSettings.crmAttachmentToken || _throw('crmAttachmentToken missing from custom settings')
+              entity_table: customSettings.entityTable || error('entityTable'),
+              crm_attachment_token: customSettings.crmAttachmentToken || error('crmAttachmentToken')
             }]
           });
-        },
-
-        /**
-         * Uploads all files in queue updating with additional form data
-         *
-         * @param {Object} fileUploader instance of FileUploader that was created by the service
-         * @param {Object} additionalFormData that has keys like entityID or description
-         *
-         * @return {Promise} that resolves to result if successful else error
-         */
-        uploadAll: function (fileUploader, additionalFormData) {
-          var deferred = $q.defer(),
-            results = [];
 
           /**
-          * FileUploader callback that gets fired before each file item gets
-          * uploaded, useful to insert additional form params like entityId
-          * which are not available at the time of uploader creation.
-          *
-          * @param {Object} item file item being uploaded
-          */
-          fileUploader.onBeforeUploadItem = function (item) {
-            for (key in additionalFormData) {
-              var snakeCaseKey = _.snakeCase(key);
-              item.formData.push({
-                snakeCaseKey: additionalFormData[key]
-              });
-            }
-          };
+           * Uploads all files in queue updating with additional form data.
+           * Wraps the default `uploadAll` in a custom method so we can
+           * return a promise
+           *
+           * @param {Object} additionalFormData that has keys like entityID or description
+           *
+           * @return {Promise} that resolves to result if successful else error
+           */
+          uploader.uploadAll = (function () {
+            oldUploadAll = uploader.uploadAll;
 
-          /**
-          * FileUploader callback that gets fired after each file item gets
-          * uploaded, useful to obtain result of upload. The result of each
-          * upload are aggregated in local results array.
-          *
-          * @param {Object} item file item being uploaded
-          * @param {Object} response obtained from server for upload
-          */
-          fileUploader.onCompleteItem = function (item, response) {
-            results.push(response);
-          };
+            return function (additionalFormData) {
+              /**
+               * FileUploader callback that gets fired before each file item gets
+               * uploaded, useful to insert additional form params like entityId
+               * which are not available at the time of uploader creation.
+               *
+               * @param {Object} item file item being uploaded
+               */
+              uploader.onBeforeUploadItem = function (item) {
```

Comment:
Please always prefer a functional approach if possible, make it for a terser code
```js
uploader.onBeforeUploadItem = function (item) {
  _.each(additionalFormData, function (val, key) {
    item.formData[_.snakeCase(key)] = val;
  });
};
```